### PR TITLE
Submission of snippets

### PR DESCRIPTION
**Pull request by Akorax**
Snippety ship.

### CODE REVIEW

File: .vscode/manic-miners.code-snippets [hunk 1]
```diff
@@ -3,16 +3,28 @@
...
-	// 		"$2"
+	// 		"console.log('');",
+	// 		""
 	// 	],
 	// 	"description": "Log output to console"
 	// }
-}
\ No newline at end of file
+	"When": {
+		"scope": "javascript,typescript",
```
Comment (cspotcode) [inline .vscode/manic-miners.code-snippets]: As discussed on Discord, we'll need to remove these "scope" entries.  Eventually we can have a "manicminerslevel" language which can go here.

I know you already know this, but putting a comment shows how we can discuss changes on github.

File: .vscode/manic-miners.code-snippets [hunk 2]
```diff
@@ -15,4 +15,225 @@
...
+	// --------------------------- OCCURENCE -----------------------------------------------------
+	"When": {
+		"prefix": "when()",
+		"body": "when(${1:Trigger})[${2:Event}]",
+		"description": "Writes when and put the cursor inside the parenthesis."
+	},
+	"If": {
+		"prefix": "if()",
+		"body": "if(${1:Trigger})[${2:Event}]",
+		"description": "Writes If and put the cursor inside the parenthesis."
```
Comment (cspotcode) [inline .vscode/manic-miners.code-snippets:27]: Should we update these descriptions to focus on the meaning of `When` and `If`, to help the scripter understand what they're doing?

File: .vscode/manic-miners.code-snippets [hunk 3]
```diff
@@ -15,4 +15,225 @@
...
+		"description": "Call the corresponding event when a wall at ROW,COLUMN is changed in any way. This includes reinforcing, shoveling once, drilling, finding a cave etc. Optionally add a ID requirement for the trigger to fire only if the tile is changed to a specific tile ID."
+	},
+	"Reinforce": {
+		"prefix": "TriggerReinforce",
+		"body": "reinforce:${1:row},${2:column}",
+		"description": "Call the corresponding event when a wall at ROW,COLUMN is reinforced by any unit."
+	},
+	"Time": {
+		"prefix": "TriggerTime",
+		"body": "time:${1:row},${2:column}",
```
Comment (cspotcode) [inline .vscode/manic-miners.code-snippets:114]: update placeholders to get rid of row, column.

**Review by cspotcode — CHANGES_REQUESTED** [review]
I think I found a typo.  Also made a suggestion about some descriptions.  I technically didn't look at *all* of it, but this looks like a great start.

File: .vscode/manic-miners.code-snippets (hunk 2 above)
Comment (Akorax) [reply .vscode/manic-miners.code-snippets:27]: I think very few will look at the descriptions. If they do they probably do it on the wiki where the most updated info is.

Comment (cspotcode) [reply .vscode/manic-miners.code-snippets:27]: That's fair, but these descriptions are still inconsistent with the others.  This one describes what is being typed, whereas the others focus on what the code means in the game.

**Comment by cspotcode** [comment]
These changes have been incorporated via https://github.com/cspotcode/ManicMinersScripting/blob/master/vscode-extension/generate-code-snippets.js

**Pull request closed without merging**